packages/backend/src/backend_id_scoped_stable_backend_identifiers.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  BackendIdentifier,
+  StableBackendIdentifiers,
+} from '@aws-amplify/plugin-types';
+import { createHash } from 'crypto';
+
+const HASH_LENGTH = 20;
+
+/**
+ * Gets a stable hash value derived from BackendIdentifier
+ */
+export class BackendIdScopedStableBackendIdentifiers
```

Comment:
This should be something like
```
export class BackendIdScopedStableHash
  implements StableHash
```
i.e. the classes here do not contain or represent `BackendIdentifiers` but `Hash` and the impl is a `BackendIdScoped`

Reply:
That's what it was before, but we changed the name to leave room for future extensibility. See comment for context: https://github.com/aws-amplify/amplify-backend/pull/1213#discussion_r1559767167